spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolver.java

```diff
@@ -32,50 +37,95 @@
 		JarFile.registerUrlProtocolHandler();
+		try {
+			Resource[] globalResources = new PathMatchingResourcePatternResolver(ApplicationConfigurationMetadataResolver.class.getClassLoader()).getResources(DATAFLOW_PROPERTIES);
+			loadWhiteLists(globalResources, globalWhiteListedClasses, globalWhiteListedProperties);
+		}
+		catch (IOException e) {
+			throw new RuntimeException("Error reading global white list of configuration properties", e);
+		}
+	}
+
+	public List<ConfigurationMetadataProperty> listProperties(Resource app) {
+		return listProperties(app, false);
 	}
 
 	/**
-	 * Return metadata about configuration properties (as groups) that are documented via
+	 * Return metadata about configuration properties that are documented via
 	 * <a href="http://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html">
-	 * Spring Boot configuration metadata</a> and visible in an application.
-	 *
-	 * @param applicationResource a Spring Cloud Stream application resource; typically a Boot uberjar,
-	 * but directories are supported as well
+	 * Spring Boot configuration metadata</a> and visible in an app.
+	 * @param app a Spring Cloud Stream app; typically a Boot uberjar,
+	 *            but directories are supported as well
 	 */
-	public List<ConfigurationMetadataGroup> listPropertyGroups(Resource applicationResource) {
-		List<ConfigurationMetadataGroup> result = new ArrayList<>();
-		ClassLoader classLoader = null;
+	public List<ConfigurationMetadataProperty> listProperties(Resource app, boolean exhaustive) {
+		List<ConfigurationMetadataProperty> result = new ArrayList<>();
+		ClassLoader moduleClassLoader = null;
 		try {
-			File applicationFile = applicationResource.getFile();
-			Archive archive = applicationFile.isDirectory() ? new ExplodedArchive(applicationFile)
-					: new JarFileArchive(applicationFile);
-			classLoader = createClassLoader(archive);
+			File moduleFile = app.getFile();
+			Archive archive = moduleFile.isDirectory() ? new ExplodedArchive(moduleFile) : new JarFileArchive(moduleFile);
+			moduleClassLoader = createClassLoader(archive);
+			ResourcePatternResolver moduleResourceLoader = new PathMatchingResourcePatternResolver(moduleClassLoader);
+
+
+			Collection<String> whiteListedClasses = new HashSet<>(globalWhiteListedClasses);
+			Collection<String> whiteListedProperties = new HashSet<>(globalWhiteListedProperties);
+
+			loadWhiteLists(moduleResourceLoader.getResources(DATAFLOW_PROPERTIES), whiteListedClasses, whiteListedProperties);
+
 			ConfigurationMetadataRepositoryJsonBuilder builder = ConfigurationMetadataRepositoryJsonBuilder.create();
-			ResourcePatternResolver applicationResourceLoader = new PathMatchingResourcePatternResolver(classLoader);
-			for (Resource r : applicationResourceLoader.getResources("classpath*:/META-INF/*spring-configuration-metadata.json")) {
+			for (Resource r : moduleResourceLoader.getResources(CONFIGURATION_METADATA_PATTERN)) {
```

Comment:
I was trying to use `additional-spring-configuration-metadata.json` to override the advertised default value for maxMessages (it needs to be 1 or -1 depending on the source) so that change was not picked up of course.  Is there some boot class we can delegate to that understands how to do the merge of these two categories of file?